interfaces/backends/backends.go

```diff
@@ -33,6 +35,14 @@ import (
 	apparmor_sandbox "github.com/snapcore/snapd/sandbox/apparmor"
 )
 
+func isContainer() bool {
+	cmd := exec.Command("systemd-detect-virt", "--container", "--quiet")
```

Comment:
we are doing this from other places, maybe time to have something in osutil?

Reply:
Yes, I saw we check for non-contianer virt as well. I'll have a look at a common helper used in both places.

Reply:
I've added a helper but one remaining call site is hard to adjust (squashfs) as it introduces dependency loops. I think this is fine for now. Without forcing a package split to resolve that.